client/src/components/layouts/dashboard/RecentTransaction.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+import { useStyles } from '../../themes/dashboardStyles/dashboardStyle';
+import { TableContainer, Table, Typography } from '@material-ui/core';
+import RecentItems from '../dashboard/RecentItems';
+
+const RecentTransaction = ({ receipts }) => {
+  const classes = useStyles();
+  const [recent, setRecent] = useState([]);
+
+  useEffect(() => {
+    const getRecent = (receipts) => {
+      let recentReceipts;
+      if (receipts !== [] && receipts.length > 20) {
+        recentReceipts = receipts.slice(0, 20);
+      } else {
+        recentReceipts = receipts;
+      }
+      setRecent(recentReceipts);
+    };
+    getRecent(receipts);
+    // eslint-disable-next-line
+  }, [receipts]);
+
+  return (
+    <div>
+      {recent.length > 0 ? (
+        <TableContainer>
+          <Table className={classes.table} aria-label="simple table">
+            {recent !== [] &&
```

Comment:
```suggestion
            {recent &&
```

instead of comparing it to an empty Array, you can either just ask if it's a Truthy value (as shown above, this is good if you are not sure if it's undefined or an array) or, you can compare the length recent.length === 0 if you are sure it's an Array

the reason is when you create a new Array you're creating a new object into the code and may be made for maintainability and also memory storage. it's small and minimal, best for best coding practice :)